lms/djangoapps/discussion/rest_api/views.py

```diff
@@ -109,6 +112,73 @@ def get(self, request, course_id):
         return Response(get_course(request, course_key))
 
 
+class CourseUsersView(DeveloperErrorViewMixin, APIView):
+    """
+    **Use Cases**
+
+        Fetch statistics about a user's activity in a course.
+
+    **Example Requests**:
+
+        GET /api/discussion/v1/courses/course-v1:ExampleX+Subject101+2015/activity_stats?order_by=activity
+
+    **Response Values**:
+
+
+    **Example Response**
+    ```json
+    {
+        "pagination": {
+            "count": 3,
+            "next": null,
+            "num_pages": 1,
+            "previous": null
+        },
+        "results": [
+            {
+                "active_flags": 3,
+                "inactive_flags": 0,
+                "replies": 13,
+                "responses": 21,
+                "threads": 32,
+                "username": "edx"
+            },
+            {
+                "active_flags": 1,
+                "inactive_flags": 0,
+                "replies": 6,
+                "responses": 8,
+                "threads": 13,
+                "username": "honor"
+            },
+            ...
+        ]
+    }
+    ```
+    """
+
+    authentication_classes = (
+        JwtAuthentication,
+        BearerAuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    permission_classes = (permissions.IsAuthenticated, permissions.IsAdminUser)
+
+    def get(self, request, course_key_string):
+        """Implements the GET method as described in the class docstring."""
+        form = UserCourseStatsForm(request.query_params)
+        if not form.is_valid():
+            raise ValidationError(form.errors)
+        data = get_course_discussion_user_stats(
+            request,
+            course_key_string,
+            form.cleaned_data['page'],
+            form.cleaned_data['page_size'],
+            UserOrdering(form.cleaned_data.get('order_by', None) or UserOrdering.BY_FLAGS),
```

Comment:
If we add a default, then this code simply becomes `frm_query_string.cleaned_data.get('order_by')`

Reply:
The issue is that the default is different for different types of users. For moderators etc it should default to ordering by flags, and for regular users it should default to activity.

Reply:
thanks.